metpy/calc/indices.py

```diff
@@ -305,3 +305,63 @@ def significant_tornado(sbcape, surface_based_lcl_height, storm_helicity_1km, sh
     return ((sbcape / (1500. * units('J/kg'))) *
             surface_based_lcl_height *
             (storm_helicity_1km / (150. * units('m^2/s^2'))) * shear_6km)
+
+
+@exporter.export
+@check_units('[pressure]', '[speed]', '[speed]', '[length]', '[speed]', '[speed]')
+def critical_angle(pressure, u, v, heights, stormu, stormv):
+    r"""Calculate the critical angle.
+
+    The critical angle is the angle between the 10m storm-relative inflow vector
+    and the 10m-500m shear vector. A critical angle near 90 degrees indicates
+    that a storm in this environment on the indicated storm motion vector
+    is likely ingesting purely streamwise vorticity into its updraft, and Esterheld and
+    Giuliano (2008) showed that significantly tornadic supercells tend to occur in environments
+    with critical angles near 90 degrees.
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        Pressures from sounding.
+    u : `pint.Quantity`
+        U-component of sounding winds.
+    v : `pint.Quantity`
+        V-component of sounding winds.
+    heights : `pint.Quantity`
+        Heights from sounding.
+    stormu : `pint.Quantity`
+        U-component of storm motion.
+    stormv : `pint.Quantity`
+        V-component of storm motion.
+
+    Returns
+    -------
+    `pint.Quantity'
+        critical angle in degrees
+
+    """
+    # Convert everything to m/s
+    u = u.to('m/s')
```

Comment:
Is converting to 'm/s' just to make sure they all have the same units? Otherwise, I think it should just flow through without problems. (Emphasis on *should*.)

Reply:
That was the idea. I just went in and fixed it.

Reply:
To be clear, I wasn't saying that was an invalid use case. I get the feeling that if you hand it u,v and stormu, stormv in different units, things will break right now. Can you double check?

Reply:
Sure! I just ran several different cases through it with different combinations of kt and m/s for units on the storm motion, and it appears that everything works as long as both components of the storm motion have the same units, regardless of what the units are for the environmental wind profile.

Reply:
Ok, then I would put back the conversion to 'm/s' so that it works in the case of things coming in different units. I could see someone having winds in m/s and storm motion in knots. What's the point of unit support if we can't handle that? Bonus points if you add a test case for that.